### PR TITLE
[JBPM-10075] - PostgreSQL validation fails at Hibernate after upgrade to Quarkus 2.7.5

### PR DESCRIPTION
**Pull request by afalhambra**
**[JBPM-10075](https://issues.redhat.com/browse/JBPM-10075)**: PostgreSQL validation fails at Hibernate after upgrade to Quarkus 2.7.5

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>

</details>

### CODE REVIEW

**Comment by mareknovotny** [comment]
please prepare also backport PR for 7.67.x ;)

**Comment by afalhambra** [comment]
> please prepare also backport PR for 7.67.x ;)

sure, just waiting for the approval before raising the backport

**Comment by antncl** [comment]
The PR contains an empty file upgrade-scripts/postgresql/rhpam-7.12-to-7.13.sql - was this intentional?

File: src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+        assertThat(results.get(0).getLogs(), containsInAnyOrder(report.getLogs().toArray()));
+
+        migrationService.delete(migration.getId());
+
+        verify(mockAdminServicesClient, times(1)).migrateProcessInstance(anyString(), anyLong(), anyString(), anyString(), anyMap());
+    }
+
+    protected void createSchema() throws IOException, SQLException {
+        scriptRunner.executeScriptRunner(INITIAL_DB_SCRIPTS_RESOURCE_PATH, ScriptFilter.create());
+    }
```
Comment (MarianMacik) [inline src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java]:
```suggestion
    protected void createSchema() throws IOException, SQLException {
        scriptRunner.executeScriptRunner(DB_CREATE_SCRIPTS_RESOURCE_PATH, ScriptFilter.create());
    }
```

File: src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    protected void createInitialSchema() throws IOException, SQLException {
+        //create PIM very first schema version
+        scriptRunner.executeScriptRunner(INITIAL_DB_SCRIPTS_RESOURCE_PATH, ScriptFilter.create());
+    }
+
+    protected void dropFinalSchemaAfterUpgrade() throws IOException, SQLException {
+        //drop schema
+        //need to drop constraints from initial version first
+        scriptRunner.executeScriptRunner(INITIAL_DB_SCRIPTS_RESOURCE_PATH, ScriptFilter.drop());
+        scriptRunner.executeScriptRunner(DB_CREATE_SCRIPTS_RESOURCE_PATH, ScriptFilter.drop());
```
Comment (MarianMacik) [inline src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java]: Final schema is equal to the create script (current version), so why do we have to drop both? Also, why do we have to run the initial drop script first?

File: src/test/java/org/kie/processmigration/test/persistence/scripts/util/SQLScriptUtil.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+     */
+    private static final String DELIMITER_MSSQL = "GO";
+    /**
+     * Delimiter used in PostGreSQL database systems.
+     */
+    private static final String DOLLAR_QUOTED_BLOCK = "\\$.*\\$";
+    /**
+     * Delimiter used in DB2 BPMS 6.0-to-6.1 upgrade scripts.
+     */
+    private static final String DELIMITER_ALTER = "@";
```
Comment (MarianMacik) [inline src/test/java/org/kie/processmigration/test/persistence/scripts/util/SQLScriptUtil.java]: Do we need all of these delimiters here? I don't think some of those are used.

**Review by MarianMacik — COMMENTED** [review]
plus one more as Ant.
> The PR contains an empty file upgrade-scripts/postgresql/rhpam-7.12-to-7.13.sql - was this intentional?

File: src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java (hunk 1 above)
Comment (afalhambra) [reply src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java]: thank you!

File: src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java (hunk 2 above)
Comment (afalhambra) [reply src/test/java/org/kie/processmigration/persistence/AbstractScriptsBaseTest.java]: Yeah, you're right, only "create script resources" (current version) would be enough.

File: src/test/java/org/kie/processmigration/test/persistence/scripts/util/SQLScriptUtil.java (hunk 3 above)
Comment (afalhambra) [reply src/test/java/org/kie/processmigration/test/persistence/scripts/util/SQLScriptUtil.java]: Done

**Comment by afalhambra** [comment]
> The PR contains an empty file upgrade-scripts/postgresql/rhpam-7.12-to-7.13.sql - was this intentional?

fixed. Thanks for noticing

File: frontend/package-lock.json [hunk 4]
```diff
@@ -1263,8 +1263,6 @@
     },
     "node_modules/@kie/lock-treatment-tool": {
       "version": "0.1.0",
-      "resolved": "https://registry.npmjs.org/@kie/lock-treatment-tool/-/lock-treatment-tool-0.1.0.tgz",
-      "integrity": "sha512-KkUPkdWdjfp8T9EmNybGHAbvXdxdOcotgG21Lyk92e4ASFb/L0x5nJMOCUdX2ras5hD39TIqT4R2v8Z56UzDxA==",
```
Comment (MarianMacik) [inline frontend/package-lock.json]: Probably this can be omitted from the PR.

**Review by MarianMacik — APPROVED** [review]
Looks good, just `package-lock.json` is not needed. Thanks!

File: frontend/package-lock.json (hunk 4 above)
Comment (afalhambra) [reply frontend/package-lock.json]: Thanks for noticing!

**Comment by afalhambra** [comment]
@AntStephenson @mareknovotny @rgdoliveira could you please provide your review? 
thanks

**Review by ruromero — APPROVED** [review]

**Review by mareknovotny — APPROVED** [review]

**Review by antncl — APPROVED** [review]

**Pull request merged**